src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,48 +3,174 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+
+    private int size;
+    private Node<T> head;
+    private Node<T> tail;
+
+    public MyLinkedList() {
+        head = null;
+        tail = null;
+        size = 0;
+    }
+
+    public MyLinkedList(T element) {
+        Node<T> firstNode = new Node<>(element);
+        head = tail = firstNode;
+        size = 1;
+    }
+
+    private static class Node<T1> {
+        T1 storedItem;
+        Node<T1> previous;
+        Node<T1> next;
+
+        public Node(T1 element) {
+            storedItem = element;
+            previous = null;
+            next = null;
+        }
+    }
+
+    public void checkBounds(int index) {
+        if (index < 0 || index >= size) {
+            throw new IndexOutOfBoundsException("Index is out of LinkedList bonds");
+        }
+    }
+
+    public void nodeInsertion(Node<T> previousNode, Node<T> nodeToInsert, Node<T> nextNode) {
```

Comment:
```suggestion
    private void insertNode(Node<T> targetNode, T value) {
```
you can get all other Nodes you need from that `targetNode` (before which you need to insert the new Node)

Reply:
done